### ISSUE TITLE
Press enter during "pick Object" dialogue

### ISSUE THREAD

**Issue by Wackytoaster** [Scirra/Construct-3-bugs#1139]
## Problem description

From C2 I´m used to navigate through most of the Condition and Event creation with keyboard. In C2 pressing 'enter' in this window does the next logical operation (letting me pick the object). In C3 a popup will tell me that I have to pick an object (duh)

## Attach a .c3p

🐰 

## Steps to reproduce

![snip](https://user-images.githubusercontent.com/31101309/34586380-107723d0-f1a4-11e7-815d-78b10345e333.PNG)
1. Go to this window (or any window with such an object picker)
2. Press Enter

## Observed result

A popup tells me to select an Object.

## Expected result

Open the select object dialogue.

## Affected browsers

* Chrome: yes
* Firefox: -
* Edge: -
* Safari: -

## System details

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 63.0.3239.84
Browser engine: Blink
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.84 Safari/537.36
C3 release: r75
Language setting: en-US

Local storage
Storage quota (approx): 14 gb
Storage usage (approx): 99 mb (0.7%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 1070 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
In C3 'Enter' is the shortcut to press 'Done'. You can press spacebar to "click" a highlighted button, which seems to work fine. Is it OK to simply use spacebar instead? If we override 'enter' to activate that button instead, then the shortcut to use 'Enter' as 'Done' only sometimes works (depending on if that button is there or not), which would be inconsistent.

**Comment by Wackytoaster** [comment]
Hmm, I see. I guess using Spacebar is something to get used to, but defenitly ok. It would be nice to know such handy shortcuts beforehand. ;)

The idea I proposed was **not** to override 'enter'. What I meant was to pick the next logical thing to do when you press 'enter'. In the picture I posted, there is nothing else to do besides picking an object. (I´m not counting going back, since I wouldn´t try to press 'enter' to go back.) I cannot proceed if no object is picked, so if I press 'enter' in that situation, my most likely intent was to select an object. Once the object is selected, pressing 'enter' will still press 'done'. If no button is present, it will also be 'done'.

It´s essentially like when a game tells you to press "Start", and you press 'A' and it starts anyway, because there are no other options that make sense. But as I said, pressing Spacebar is fine as long as I know I have that option.
